### DISCUSSION TITLE
Improving performance auto-reload of FastAPI app with lifespan events

### DISCUSSION THREAD

**Discussion by havardthom** [Kludex/uvicorn#1887] [Q&A]
I'm running a FastAPI app using uvicorn with auto-reload and it takes quite some time for the app to run shutdown event and startup event on every auto-reload. Does anyone have any tips/tricks regarding speeding up auto-reload of a FastAPI app?

Ideally, it would be great with an option where lifespan events only run on first startup and shutdown and does not need to run on every auto-reload. In my case the startup event does a lot of service initializations and assigns to `app.state` (because factory pattern), so somehow caching `app.state` would alleviate a lot, but that is very specific to my usecase.

**Comment by y2kbugger** [comment]
https://ipython.readthedocs.io/en/stable/config/extensions/autoreload.html#autoreload

existence proof of the concept.
